pvlib/iotools/bsrn.py

```diff
@@ -162,8 +162,7 @@ def get_bsrn(station, start, end, username, password,
     end = pd.to_datetime(end)
 
     # Generate list files to download based on start/end (SSSMMYY.dat.gz)
-    filenames = pd.date_range(
-        start, end.replace(day=1) + pd.DateOffset(months=1), freq='1M')\
+    filenames = pd.date_range(start, end, freq='1MS')\

```

Comment:
I think this doesn't work if the inputs are midway through a month.  Compare:

```python
In [1]: import pandas as pd
   ...: 
   ...: start = pd.to_datetime("2021-01-10")
   ...: end = pd.to_datetime("2021-03-03")

In [2]: pd.date_range(start, end.replace(day=1) + pd.DateOffset(months=1), freq='1M')
Out[2]: DatetimeIndex(['2021-01-31', '2021-02-28', '2021-03-31'], dtype='datetime64[ns]', freq='M')

In [3]: pd.date_range(start, end, freq='1MS')
Out[3]: DatetimeIndex(['2021-02-01', '2021-03-01'], dtype='datetime64[ns]', freq='MS')
```

Reply:
I copied that over from the original PR, but it is indeed flawed.

I think this should work:
```python
pd.date_range(start.date().replace(day=1), end, freq='1MS')
```